Source/LibationWinForms/grid/ProductsGrid.cs

```diff
@@ -225,5 +244,102 @@ public void Filter(string searchString)
 		#region DataGridView Macro
 		private GridEntry getGridEntry(int rowIndex) => _dataGridView.GetBoundItem<GridEntry>(rowIndex);
 		#endregion
+
+		#region Column Customizations
+
+		protected override void OnVisibleChanged(EventArgs e)
+		{
```

Comment:
Do all column customization in OnVisibleChanged. Weird things can happen in some edge cases if these tasks are done inside the constructor or OnLoad, and since this is a Control not a Form, we don't have access to a Shown event.